chore(ATH Docker Image) build and deploy with infra.ci.jenkins.io

dduportal opened the pull request:
Ref. https://github.com/jenkins-infra/helpdesk/issues/3084

This PR adds a 2nd pipeline which goal is to build/test/deploy the ATH Docker image on the Jenkins controller infra.ci.jenkins.io.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- ~[ ] Link to relevant pull requests, esp. upstream and downstream changes~
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

timja commented:
The checks are a bit confusing in that there's soo many for the dockerfile, and then the overall build status is `continuous-integration/jenkins/pr-merge` which is for the docker build and not the main build.

Would it be possible to turn off the stage builds and configure the overall build status to be `Docker build` ?

timja commented on src/main/resources/ath-container/Dockerfile:
```diff
@@ -3,8 +3,11 @@
 
 FROM centos:7
-MAINTAINER ogondza@gmail.com
+LABEL maintainer="ogondza@gmail.com"
```
```suggestion
```

dduportal commented:
> The checks are a bit confusing in that there's soo many for the dockerfile, and then the overall build status is `continuous-integration/jenkins/pr-merge` which is for the docker build and not the main build.
> 
> Would it be possible to turn off the stage builds and configure the overall build status to be `Docker build` ?

Yes of course, nice suggestion!

Thanks for the review.

This PR stays in draft, because the pipeline-library needs to be updated to allow specifying the registry (and adding the correct credentials) because `jenkinsciinfra` is used by default and there are no way to override it today. Should be easy.


I'll rebase this PR and set it as "ready to review" once done.

dduportal commented:
> The checks are a bit confusing in that there's soo many for the dockerfile, and then the overall build status is `continuous-integration/jenkins/pr-merge` which is for the docker build and not the main build.
> 
> Would it be possible to turn off the stage builds and configure the overall build status to be `Docker build` ?

@timja I can't easily find a way to disable the status check "per stage". I can find how to publish a custom one from pipeline though.

I'me playing around with the mulitbranch job configuration (manual changes for now) on infra.ci and just set it up like this:

<img width="1377" alt="Capture d’écran 2022-08-23 à 15 33 18" src="https://user-images.githubusercontent.com/1522731/186171928-12e1f59e-c790-40ea-b8cd-cb802a556005.png">

But it seems that the checks are kept over time. Do you have a way to clean them up or should I close this PR and open a new one with this setup?

timja commented:
if you push a new commit it should get rid of the old ones.

The stage one is probably ok, as long as the 'default' looking link doesn't point to infra.ci which most people can't access

dduportal commented:
> if you push a new commit it should get rid of the old ones.

Alas it does not :( I've rebased and updated the branch: still the same checks. It's not clear how to disable it.
Even renaming the check did not had any effect, I'm not sure if I did something wrong or what.

> 'default' looking link

Which one exactly (screenshot to clarify)?

dduportal commented:
> point to infra.ci which most people can't access

I'm confused between the "status checks" on the "Conversation" tabs and the "Checks" on the "Checks" tabs.
It seems that my changes of the job configuration are impacting the "Checks" tabs, which I would like to enable again so that users would be able to get the lint/build logs.

But I don't understand how to tune the "status checks" though 🤔
